### ISSUE TITLE
Closing the last open file, then hiding and showing the explorer reopens the file

### ISSUE THREAD

**Issue by JosiahWitt** [Microsoft/vscode#68132]
- VSCode Version: 1.31.0
- OS Version: Linux x86 (Solus)

Steps to Reproduce:

1. Close all open files
2. Open a file or two from the explorer
3. Close the files (they remain selected in the explorer)
4. Hide the explorer (either click on the explorer icon, CTRL+B, or switch to searching, extensions, etc.)
5. Show the explorer
6. The most recently closed file is now opened

This behavior has changed from the previous release. In the previous release, the file was deselected from the explorer when it was closed. In 1.31.0, the file remains selected, which gives the behavior described above.

I prefer the previous behavior, as usually, I don't want to reopen the most recently closed file. It causes extra visual noise.

Some examples of past related issues are: https://github.com/Microsoft/vscode/issues/998 and https://github.com/Microsoft/vscode/issues/6922

Does this issue occur when all extensions are disabled?: Yes

![Example of behavior](https://user-images.githubusercontent.com/7329219/52428262-9bae5080-2acf-11e9-8dad-838d77936996.gif)

**Comment by JosiahWitt** [comment]
I'm curious what Issue is tracking this? The one that references it was closed as a duplicate of this one. Thanks!

**Comment by isidorn** [comment]
Some old issue assigned to me.
Try to find it, you would do the same as me. The point in that issue is that this is designed behavior.

**Comment by JosiahWitt** [comment]
Oh, it is designed behavior? The behavior changed from version 1.30, and it has been driving me nuts! I don't mind creating a PR if you point me to relevant code. Thanks!

**Comment by isidorn** [comment]
My mistake, I have wronlgy read your description. Reopening and assigning to this milestone to investigate.

**Comment by JosiahWitt** [comment]
No worries! Something else that may be related is the case that if you select a file in the explorer tree, and then click on a different tab, the active file is not highlighted in the tree. Version 1.30 would always highlight the active file in the explorer tree.

Let me know if you need any more info, or if I can help with writing the bugfix! Thanks!

![Example](https://user-images.githubusercontent.com/7329219/52969400-3c2d2c00-337e-11e9-8f78-6c62c405114e.gif)

**Comment by isidorn** [comment]
Fixed. You can try it out in wednesday vscode insiders

**Comment by JosiahWitt** [comment]
Great! Thank you! I'll check it out!

**Comment by JosiahWitt** [comment]
@isidorn The main issue I reported seems to be fixed in the insiders build today! :tada: The related bug I mentioned in a comment above still remains. Should I open another Issue for that bug? Thank you!

**Comment by isidorn** [comment]
Yes, please open a new issue.

**Comment by JosiahWitt** [comment]
Alright! I created #69152. Thanks again!

**Comment by jacobweber** [comment]
Tried this on the insiders build from today.

It works if `explorer.autoReveal` is true (the default). When you close the last file, it de-selects the file in the Explorer, so it's not opened the next time you switch to Explorer.

But if `explorer.autoReveal` is false, it doesn't change the Explorer selection when you close a file (by design). And switching back to Explorer re-opens the selected file. So the confusing behavior is still there.

I think there should be a preference that stops this behavior; e.g. "don't open selected file when switching to Explorer".

FYI, I also opened #52638 about this a while back.
